source/run-command/run-command.rst

```diff
@@ -187,8 +193,131 @@ Drivers MUST document the behavior of RunCommand if a ``maxTimeMS`` field  is al
+A driver can expect that the result from running this command will return a document with a ``cursor`` field and MUST provide the caller with a language native abstraction to continue iterating the results from the server.
+If the response from the server does not include a ``cursor`` field the driver MUST throw an error either before returning from ``runCursorCommand`` or upon first iteration of the cursor.
+
+High level RunCursorCommand steps:
+
+* Run the cursor creating command provided by the caller and retain the ClientSession used as well as the server the command was executed on.
+* Create a local cursor instance and store the ``firstBatch``, ``ns``, and ``id`` from the response.
+* When the current batch has been fully iterated, execute a ``getMore`` using the same server the initial command was executed on.
+* Store the ``nextBatch`` from the ``getMore`` response and update the cursor's ``id``.
+* Continue to execute ``getMore`` commands as needed when the caller empties local batches until the cursor is exhausted.
+
+Driver Sessions
+"""""""""""""""
+
+A driver MUST create an implicit ClientSession if none is provided and it MUST be attached for the duration of the cursor's lifetime.
+All ``getMore`` commands constructed for this cursor MUST send the same ``lsid`` used on the initial command.
+A cursor is exhausted when the server reports a cursor is equal to zero.
+When the cursor is exhausted the client session MUST be ended and the server session returned to the pool as early as possible rather than waiting for a caller to completely iterate the final batch.
+
+* See Drivers Sessions' section on `Sessions and Cursors <https://github.com/mongodb/specifications/blob/master/source/sessions/driver-sessions.rst#sessions-and-cursors>`_
+
+Server Selection
+""""""""""""""""
+
+RunCursorCommand MUST support a ``readPreference`` option that MUST be used to determine server selection.
+The selected server MUST be used for subsequent ``getMore`` commands.
+
+Load Balancers
+""""""""""""""
+
+When in ``loadBalanced`` mode, a driver MUST pin the connection used to execute the initial operation, and reuse it for subsequent ``getMore`` operations.
+
+* See Load Balancer's section on `Behaviour With Cursors <https://github.com/mongodb/specifications/blob/master/source/load-balancers/load-balancers.rst#behaviour-with-cursors>`_
+
+Iterating the Cursor
+""""""""""""""""""""
+
+Drivers MUST provide an API, typically, a method named ``next()``, that returns one document per invocation.
+If the cursor's batch is empty and the cursor id is nonzero, the driver MUST perform a ``getMore`` operation.
+
+Executing GetMores
+""""""""""""""""""
+
+The cursor API returned to the caller MUST offer an API to configure ``batchSize``, ``maxTimeMS``, and ``comment`` that are sent on the ``getMore`` command.
```

Comment:
> Does that context steer your thinking towards leaving comment configurable or shall we prohibit / leave it up to drivers?

I don't see the harm in making it configurable. To my knowledge, the server doesn't place any restrictions on the field. If it's absent the original command's comment applies internally, and presumably there's no problem with changing it between `getMore` commands (perhaps only possible if the driver exposes the API on the CommandCursor object).

Probably better to specify it instead of leaving it undefined and inviting drivers to cook up conflicting implementations. This way we can also ensure there is test coverage.

> I'll pick a lane which is that users should be able to change these settings at any time in the lifetime of the cursor and the next getMore after the settings have changed should respect the changes.

No argument there. Let's codify this in the spec with some MAY language so drivers know they're permitted to let applications change these settings between `getMore` commands (if their API would allow for it).

I agree that we wouldn't be able to test this. You can consider noting that in the spec and saying that drivers choosing to support that SHOULD write their own tests for it.

Reply:
Updated the language to have MAY for configuration and if so a MUST for testing.